db/session/session.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package session
+
+import (
+	"context"
+
+	"github.com/sourcenetwork/defradb/datastore"
+)
+
+type contextKey string
+
+const (
+	TxnContextKey = contextKey("txn")
```

Comment:
suggestion: To avoid allocation on the key, you can simply use
```go
type TxnContextKey struct{}
```
https://pkg.go.dev/context#WithValue